### PR TITLE
Add Path property

### LINKED ISSUE #8074: Storage: Add `Path` to both `DirectoryClient` and `FileClient`
https://apiview.azurewebsites.net/Assemblies/Review/47c9ebbff8c0458fa2e7904eae34656f#Azure.Storage.Files.DirectoryClient

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes #8074

### CODE REVIEW

File: sdk/storage/Azure.Storage.Files/tests/FileClientTests.cs [hunk 1]
```diff
@@ -57,34 +57,39 @@ public void FilePathsParsing()
...
             TestHelper.AssertCacheableProperty("file.txt", () => fileClient2.Name);
+            TestHelper.AssertCacheableProperty("dir1/file.txt", () => fileClient2.Path);
             Assert.AreEqual("file.txt", builder2.LastDirectoryOrFileName);
 
             // trailing slash
             Uri uri3 = new Uri("http://dummyaccount.file.core.windows.net/share/dir1/file.txt/");
             var builder3 = new FileUriBuilder(uri3);
             var fileClient3 = new FileClient(uri3);
             TestHelper.AssertCacheableProperty("file.txt", () => fileClient3.Name);
+            TestHelper.AssertCacheableProperty("dir1/file.txt/", () => fileClient3.Path);
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files/tests/FileClientTests.cs]: This feels a little odd - why do we kill it from the name but not the path?

**Review by tg-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files/tests/FileClientTests.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/storage/Azure.Storage.Files/tests/FileClientTests.cs]: I don't feel too strongly about this, but having a slash in the Name clearly doesn't make sense, whereas in a path it seems reasonable.

Comment (JoshLove-msft) [reply sdk/storage/Azure.Storage.Files/tests/FileClientTests.cs]: Also, it is included in the DirectoryOrFilePath property on the builder.

Comment (tg-msft) [reply sdk/storage/Azure.Storage.Files/tests/FileClientTests.cs]: I think we should probably remove trailing slashes from DirectoryOrFilePath in FileUriBuilder.  I wouldn't expect a file path to end in a slash.

Comment (JoshLove-msft) [reply sdk/storage/Azure.Storage.Files/tests/FileClientTests.cs]: Ok fixed.

**Review by amnguye — APPROVED** [review]

**Pull request merged**